### PR TITLE
ROADMAP: update the TiDB roadmap

### PR DESCRIPTION
**Pull request by yikeke**
Via: https://github.com/pingcap/docs-cn/pull/1088/ All comment addressed. @lilin90 @winkyao @morgo
PTAL.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
+    - [ ] Support Change Column Type
+    - [ ] Support Multiple DDL Operations in a Single Statement
+- [ ] Support Plugin System
+    - [ ] Support White List Plugin
+    - [ ] Support Audit Log Plugin
+    - [ ] Support RBAC Plugin
+    - [ ] Support Diagnostic Plugin
+- [ ] Support Query Tracing
+- [ ] Support Hybrid Column/Row-oriented Storage Engine
+- [ ] Support New Storage Row Format: improve performance and use less memory
```
Comment (lilin90) [inline ROADMAP.md]:
```suggestion
- [ ] Support New Storage Row Format: improve performance and reduce memory usage
```

File: ROADMAP.md [hunk 2]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
+- [ ] Support Non-integer Type of RowID
+- [ ] Transaction
+    - [ ] Reduce Read-write Conflicts
+    - [ ] Optimize Scheduling Mechanism of Transaction
```
Comment (lilin90) [inline ROADMAP.md]:
```suggestion
    - [ ] Optimize Transaction Scheduling Mechanism 
```

File: ROADMAP.md [hunk 3]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
-+ [ ] Statistics
-    - [x] Update statistics dynamically according to the query feedback
-    - [x] Analyze table automatically
-    - [x] Improve the accuracy of Row Count estimation
+    - [ ] Statistics Optimization
+    - [ ] Multi-Column Statistics
+    - [ ] Cascades Planner
+    - [ ] Plan Management
+    - [ ] SQL Tuning Advisor
+    - [ ] Robust Access Path Selection: add the heuristic rule and improve the accuracy rate of index selection in OLTP scenarios
```
[the remark this answers is not in the document]
Comment (morgo) [reply ROADMAP.md]: I think it should be "add a heuristic rule", since how to handle this problem is not finite.

File: ROADMAP.md [hunk 4]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
+    - [ ] Parallel Operators
+    - [ ] Memory Control
+    - [ ] Concurrency Control
+    - [ ] Shuffle Operators
+    - [ ] Vectorized Expression Evaluation
+    - [ ] UDF
++ [ ] SQL Features
+    - [ ] Support Views
+    - [ ] Support Window Functions
+    - [ ] Support Common Table Expression
```
Comment (morgo) [inline ROADMAP.md]:
```suggestion
    - [ ] Support Common Table Expressions
```

File: ROADMAP.md [hunk 5]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
+    - [ ] Support Hash Partitioned Table
```
Comment (morgo) [inline ROADMAP.md]:
```suggestion
    - [ ] Support Hash Partitioned Tables
```

File: ROADMAP.md [hunk 6]
```diff
@@ -11,40 +11,44 @@ This document defines the roadmap for TiDB development.
...
+    - [ ] Support utf8_general_ci Collation
```
Comment (morgo) [inline ROADMAP.md]:
```suggestion
    - [ ] Support the utf8mb4_0900_ai_ci Collation
```

**Pull request merged**